lib/utils/get-registry-ky.ts

```diff
@@ -1,29 +1,47 @@
-import ky from "ky"
+import ky from "ky";
 
-// Flag to identify if we're running inside runframe specifically
+// Flag to identify environment configuration
 declare global {
   interface Window {
-    __RUNFRAME_REGISTRY_BASE_URL__?: boolean
+    __RUNFRAME_REGISTRY_BASE_URL__?: boolean;
+    REGISTRY_API_BASE_URL?: string; // For custom API URL
   }
 }
 
 // Create a function that provides the configured ky instance
-// This ensures we check the flag every time we need the client
+// This ensures we check the flag/config every time we need the client
 export function getRegistryKy() {
-  // Check flag at execution time rather than module load time
-  const useRegistryPrefix =
-    typeof window !== "undefined" &&
-    window.__RUNFRAME_REGISTRY_BASE_URL__ === true
-
-  // Set the appropriate prefix based on the environment
-  const registryApiBaseUrl = useRegistryPrefix
-    ? "/registry"
-    : "https://registry-api.tscircuit.com"
+  // For runframe specific handling
+  const isRunframe = 
+    typeof window !== "undefined" && 
+    window.__RUNFRAME_REGISTRY_BASE_URL__ === true;
+  
+  // For custom registry API URL (can be set by the importing application)
+  const customApiUrl = 
+    typeof window !== "undefined" && 
+    window.REGISTRY_API_BASE_URL;
+  
+  // Detect if we're in a localhost environment
+  const isLocalhost = 
+    typeof window !== "undefined" && 
+    (window.location.hostname === "localhost" || 
+     window.location.hostname === "127.0.0.1");
+  
+  // Determine the registry API URL with this priority:
+  // 1. If in runframe, use "/registry"
+  // 2. If custom URL is provided, use that
+  // 3. If in localhost but not runframe, use relative "/api" path
+  // 4. Otherwise use production URL
+  const registryApiBaseUrl = 
+    isRunframe ? "/registry" :
+    customApiUrl ? customApiUrl :
+    isLocalhost ? "/api" : // Default to "/api" for other localhost environments
+    "https://registry-api.tscircuit.com";
```

Comment:
```suggestion
  const registryApiBaseUrl = 
    window.TSCIRCUIT_REGISTRY_API_BASE_URL ??
    "https://registry-api.tscircuit.com";
```

pattern for this is always the same. It's confusing to have anything except just this

Reply:
could also do this to keep nextjs compat

```suggestion
  const registryApiBaseUrl = 
    (typeof window !== undefined ? window.TSCIRCUIT_REGISTRY_API_BASE_URL : null) ??
    "https://registry-api.tscircuit.com";
```